lib/pure/asyncstreams.nim

```diff
@@ -65,6 +75,10 @@ proc finished*[T](future: FutureStream[T]): bool =
   ## no data waiting to be retrieved.
   result = future.finished and future.queue.len == 0
 
+proc failed*[T](future: FutureStream[T]): bool =
```

Comment:
how about instead introducing an API with opposite semantics `succeeded`, to avoid double negation and make it easier to write things like `doAssert fut.succeeded` etc

Reply:
Then it would be inconsistent `Future[T].failed()` API  ¯\\_ツ_/¯